src/group_theory/monoid_localization.lean

```diff
@@ -266,25 +287,335 @@ show _ * _ = _, by rw [mul_inv_left, mul_one]
   f.mk' x ⟨x, H⟩ = 1 :=
 by convert mk'_self _ _; refl
 
-@[simp, to_additive] lemma mul_mk'_eq_mk'_of_mul (x₁ x₂) (y : S) :
+@[to_additive] lemma mul_mk'_eq_mk'_of_mul (x₁ x₂) (y : S) :
   f.1 x₁ * f.mk' x₂ y = f.mk' (x₁ * x₂) y :=
 by rw [←mk'_one, ←mk'_mul, one_mul]
 
-@[simp, to_additive] lemma mk'_mul_eq_mk'_of_mul (x₁ x₂) (y : S) :
+@[to_additive] lemma mk'_mul_eq_mk'_of_mul (x₁ x₂) (y : S) :
   f.mk' x₂ y * f.1 x₁ = f.mk' (x₁ * x₂) y :=
 by rw [mul_comm, mul_mk'_eq_mk'_of_mul]
 
-@[simp, to_additive] lemma mul_mk'_one_eq_mk' (x) (y : S) :
+@[to_additive] lemma mul_mk'_one_eq_mk' (x) (y : S) :
   f.1 x * f.mk' 1 y = f.mk' x y :=
 by rw [mul_mk'_eq_mk'_of_mul, mul_one]
 
-@[simp, to_additive] lemma mk'_mul_cancel_right (x : M) (y : S) :
+@[to_additive] lemma mk'_mul_cancel_right (x : M) (y : S) :
   f.mk' (x * y) y = f.1 x :=
 by rw [←mul_mk'_one_eq_mk', f.1.map_mul, mul_assoc, mul_mk'_one_eq_mk', mk'_self, mul_one]
 
-@[simp, to_additive] lemma mk'_mul_cancel_left (x) (y : S) :
+@[to_additive] lemma mk'_mul_cancel_left (x) (y : S) :
   f.mk' ((y : M) * x) y = f.1 x :=
 by rw [mul_comm, mk'_mul_cancel_right]
 
+variables {g : M →* P}
+
+@[to_additive] lemma comp_eq_of_eq {T : submonoid P} {Q : Type*} [comm_monoid Q]
```

Comment:
This lemma needs a docstring I think

Reply:
Is there a different way of stating this? Is it a consequence of the fact that `k.1 ∘ g` is a monoid hom into a monoid such that everything in `S` has an inverse. Is this a better way to state it?

Reply:
I've added a docstring but I can't really think of a much better way of stating it - it's only used once in the file and the proof is short; I could just remove it.